src/neural/cuda/network_cudnn.cc

```diff
@@ -310,6 +310,32 @@ class CudnnNetwork : public Network {
     if (!options.IsDefault<bool>("custom_winograd"))
       use_custom_winograd_ = options.Get<bool>("custom_winograd");
 
+    // Warn if the memory required for storing transformed weights is
+    // going to exceed 60% of total video memory, force custom_winograd off
+    // if it's going to exceed 80% of memory.
+    size_t residual_single_layer_weight_size =
+        3 * 3 * kNumFilters * kNumFilters * sizeof(DataType);
+    size_t residual_weight_size =
+        residual_single_layer_weight_size * numBlocks_ * 2;
+    size_t transformed_residual_weight_size = residual_weight_size * 4;
+    if (residual_weight_size > 0.6 * deviceProp.totalGlobalMem) {
+      CERR << "Low video memory detected. You may run into OOM errors. Please "
+              "consider using a smaller network.";
+      // No hope of using custom winograd - even the fallback path might not run.
+      use_custom_winograd_ = false;
+    } else if (use_custom_winograd_) {
+      if (transformed_residual_weight_size > 0.8 * deviceProp.totalGlobalMem) {
```

Comment:
I was thinking about an additional check here to only disable the custom code if it wasn't explicitly requested by the user, allowing to overriding this (with the warning in the else branch) but this is probably overkill.

Reply:
I'm going assume that we can do such things in a follow up if they end up looking useful and just merge this for now.

Reply:
As I said, overkill. Will only consider it if we get complaints about it.